sdk/python/README.md

```diff
@@ -0,0 +1,258 @@
+# KFP-Tekton Developer Guide
+
+This document describes the development guidelines for contributing to the KFP-Tekton project. 
+Details about the required contributor license agreement (CLA) and the code review process can be found in the
+[CONTRIBUTING.md](/CONTRIBUTING.md) document.
+A quick-start guide with general setup instruction, trouble shooting guide and technical limitations can be found in
+the [SDK README](/sdk/README.md)
+
+## Table of Contents
+
+<!-- START of ToC generated by running ./scripts/mdtoc.sh sdk/python/README.md -->
+
+  - [Development Prerequisites](#development-prerequisites)
+    - [Installing Tekton](#installing-tekton)
+      - [Tekton Cluster](#tekton-cluster)
+      - [Tekton CLI](#tekton-cli)
+      - [Tekton Dashboard](#tekton-dashboard)
+  - [Origins of the KFP-Tekton Compiler Code](#origins-of-the-kfp-tekton-compiler-code)
+  - [Adding New Code](#adding-new-code)
+    - [Overriding KFP Compiler Methods](#overriding-kfp-compiler-methods)
+    - [Monkey-Patching Static KFP Compiler Methods](#monkey-patching-static-kfp-compiler-methods)
+  - [Coding Style](#coding-style)
+  - [Testing](#testing)
+    - [Unit Tests](#unit-tests)
+    - [End-to-End Tests with Tekton](#end-to-end-tests-with-tekton)
+    - [Compiler Test Report](#compiler-test-report)
+  - [License Headers](#license-headers)
+
+<!-- END of ToC generated by running ./scripts/mdtoc.sh sdk/python/README.md -->
+
+
+## Development Prerequisites
+
+1. [`Python`](https://www.python.org/downloads/): version `3.5` or later
+2. [`Kubernetes` Cluster](https://v1-15.docs.kubernetes.io/docs/setup/): version `1.15` ([required by Kubeflow](https://www.kubeflow.org/docs/started/k8s/overview/) and Tekton 0.11)
+3. [`kubectl` CLI](https://kubernetes.io/docs/tasks/tools/install-kubectl/): required to deploy Tekton pipelines to Kubernetes cluster
+4. [`Tekton` Deployment](https://github.com/tektoncd/pipeline/releases/tag/v0.11.0/): version `0.11.0` (or greater to support Tekton API version `v1beta1`), required for end-to-end testing
```

Comment:
maybe we should use v0.11.3 to support the new condition passing?

Reply:
yup, I updated it, thanks